image_manager.py

```diff
@@ -1,44 +1,94 @@
-        new_file_name = self._get_img_path(file_name, item_id, vector)
-        self.bucket.upload_file(
-            file_name, new_file_name)
-        return new_file_name
-    
-    def add_vector_image(self, file_name, item_id):
-        return self.add_image(file_name, item_id, vector=True)
-    
-    def add_png_image(self, file_name, item_id):
-        return self.add_image(file_name, item_id, vector=False)
-
-    def delete_image(self, item_id, vector=True):
-        folder = self.vector_folder if vector else self.png_folder
-        img_path = os.path.join(folder, item_id)
-        response = self.bucket.objects.filter(Prefix=img_path).delete()
-        if response and response[0]['ResponseMetadata']['HTTPStatusCode'] == 200:
+
+    def upload_file(self, bytes_file: bytes, file_name: str, bucket_name: str):
+        response = self.bucket.put_object(Body=bytes_file, Key=file_name)
+        return response
+
+    def upload_vector_file(self, file: str, item_id: str, subfolder: str = ''):
+        """
+        Keyword arguments:
+        file: a string reference to the file on your computer
+        item_id: identifying string
+        subfolder: folder path(s) if needed.
+
+        Return: image path
+        """
+        if self._is_supported_format(file, ImageConf.VECTOR_FORMATS):
+            file_ext = self._get_file_ext(file)
+            image = open(file, "rb")
+            bytes_file = image.read()
+            image_path = self._get_img_path(item_id, file_ext, subfolder)
+            self.upload_file(bytes_file, image_path, self.bucket)
+            image.close()
+            return image_path
+
+    def upload_png_file(self, file: str, item_id: str, subfolder: str = ''):
+        png = self._get_png(file)
+        image_path = self._get_img_path(item_id, '.png', subfolder)
+        self.upload_file(png, image_path, self.bucket)
+        return image_path
+
+    def delete_file(self, image_path: str):
+        """Return: True if query was successful, whether object existed or not"""
+        response = self.bucket.Object(image_path).delete()
+        if response and response['ResponseMetadata']['HTTPStatusCode'] == 204:
             return True
         return False
 
-    def _get_img_path(self, file_name, item_id, vector=True):
-        folder = self.vector_folder if vector else self.png_folder
-        ext = os.path.splitext(file_name)[1]
-        return os.path.join(folder, item_id) + ext
+    def _get_img_path(self, item_id: str, file_ext: str, subfolder: str = ''):
+        return os.path.join(subfolder, item_id) + file_ext
+
+    def _get_image_object(self, file_path: str):
+        return Image.open(file_path)
+
+    def _get_file_ext(self, file_path: str):
+        return os.path.splitext(file_path)[1].lower()
+
+    def _is_supported_format(self, file_path: str, formats: tuple):
+        file_ext = self._get_file_ext(file_path)
+        return file_ext in formats
```

Comment:
Is there a point in having the formats argument? Do you ever want to use something else than `ImageConf.VECTOR_FORMATS`?

Reply:
will likely use this function to check if other formats are supported (if someone chooses to upload a separate png file for example) thus I'll need to be able to check different formats. leaving as is